### PR TITLE
[dotnet] - Support noble distro and remove stale powershell security patch

### PR DESCRIPTION
**Pull request by gauravsaini04**
**Devcontainer Image**

* Dot Net

**_Changelog_**

* This PR tries to fix the issue where the `powershell` installation fails on variants like `ubuntu:noble`

### CODE REVIEW

File: src/dotnet/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -4,29 +4,4 @@ ENV PATH $PATH:/home/vscode/.dotnet:/home/vscode/.dotnet/tools
 
 # clear this environment variable so xml docs from NuGet packages are unpackaged. The default dotnet/sdk image sets it to 'skip'.
 # see https://github.com/dotnet/dotnet-docker/issues/2790
-ENV NUGET_XMLDOC_MODE=
-
-# Temporary: Upgrade packages due to mentioned CVEs
-# They are installed by the base image (mcr.microsoft.com/dotnet/sdk) which does not have the patch.
-# https://msrc.microsoft.com/update-guide/vulnerability/CVE-2024-30045
```
Comment (samruddhikhandale) [inline src/dotnet/.devcontainer/Dockerfile:11]: Do we no longer want any powershell patches for for any of the variants of this image?

Comment (samruddhikhandale) [reply src/dotnet/.devcontainer/Dockerfile:11]: image

Cool!

**Review by samruddhikhandale — APPROVED** [review]

File: src/dotnet/.devcontainer/Dockerfile (hunk 1 above)
Comment (gauravsaini04) [reply src/dotnet/.devcontainer/Dockerfile:11]: yes, that's right, base image already shipping with the reqd. version

**Review by samruddhikhandale — APPROVED** [review]
Awesome, thanks!

**Pull request merged**